### PR TITLE
chore: remove unnecessary patch imports and change the patch branch name

### PR DESCRIPTION
**Pull request by OGPoyraz**
## **Description**

This PR removes unnecessary files from corrupted patch introduced in the https://github.com/MetaMask/metamask-mobile/pull/9442/files#diff-83d0f6fc92e49e38dc038764255880631a481838b47b7dbaf40837264502d5e2

The reason that these lines/files added [in that PR](https://github.com/MetaMask/metamask-mobile/pull/9442/files#diff-83d0f6fc92e49e38dc038764255880631a481838b47b7dbaf40837264502d5e2) is not having a clean build in the core right before patching. 

For example if you build core in the `main` branch right now, you will have some files under helpers 
e.g `transaction-controller/dist/helpers/EtherscanRemoteTransactionSource.d.ts`. But in the [patch branch](https://github.com/MetaMask/core/compare/main...patch/mobile-transaction-controller-13-0-0) these files are under the root folder `transaction-controller/dist/EtherscanRemoteTransactionSource.d.ts`. So if you don't clean your local before patching, these files will be also included. To prevent this issue, do a `yarn build:clean` in the core package before patching in the mobile. 

We also keep the branch name as is and expect developers open PR against it, if it gets merged in to `main` mobile, then we update the patch branch.

Patch branch: https://github.com/MetaMask/core/compare/main...patch/mobile-transaction-controller-13-0-0

## **Related issues**

N/A

## **Manual testing steps**

N/A

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by tommasini — CHANGES_REQUESTED** [review]
It is intended to be removing this amount of files from the patch?

File: patches/@metamask+transaction-controller+13.0.0.patch [hunk 1]
```diff
@@ -1243,174 +1243,6 @@ index 0000000..7efd818
...
 \ No newline at end of file
-diff --git a/node_modules/@metamask/transaction-controller/dist/helpers/EtherscanRemoteTransactionSource.d.ts b/node_modules/@metamask/transaction-controller/dist/helpers/EtherscanRemoteTransactionSource.d.ts
```
Comment (tommasini) [inline patches/@metamask+transaction-controller+13.0.0.patch:1246]: For example here we are removing the .d.ts of the EtherscanRemoteTransactionSource file, was this expected?

Comment (tommasini) [reply patches/@metamask+transaction-controller+13.0.0.patch:1246]: After talking with @OGPoyraz understood that this is file is not being used on this directory, instead this file is actually leaving on the source folder and this was just not updated

**Review by tommasini — APPROVED** [review]
Amazing cleanup @OGPoyraz This looks awesome!

Just add manual steps testing for the release qa, a video showing a transactions and dapp transaction working, and the core branch that it was used to generate the patch and we are good to merge

**Comment by OGPoyraz** [comment]
> Amazing cleanup @OGPoyraz This looks awesome!
> 
> Just add manual steps testing for the release qa, a video showing a transactions and dapp transaction working, and the core branch that it was used to generate the patch and we are good to merge

Requested a regression from @sleepytanya

**Review by jpuri — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

**Comment by sleepytanya** [comment]
QA build [Bitrise](https://app.bitrise.io/build/38bb3f57-71c6-46a8-b81f-eba4562d83ee?tab=artifacts)

- [x]  Transactions - send native token origin MM
- [x]  Transactions - send native token origin dapp
- [x]  Transactions - send ERC20 token origin MM
- [x]  Transactions - send ERC20 token origin dapp
- [x]  Transactions - send ERC721 token origin MM
- [x]  Transactions - send ERC721 token origin dapp
- [x]  Transactions - speed up transaction
- [x]  Transactions - cancel transaction
- [x]  Speed up cancellation
- [x]  Tokens - import ERC20 token origin MM
- [x]  Tokens - import ERC20 token origin dapp
- [x]  Tokens - import ERC721 token origin MM
- [x]  Tokens - import ERC721 token origin dapp 
- [x]  Tokens - import ERC1155 token origin MM
- [x]  Tokens - import ERC1155 token origin dapp
- [x]  Tokens - ERC1155 transfer origin MM
- [x]  Tokens - ERC1155 batch transfer
- [x]  Tokens - approve ERC1155 token
- [x]  Tokens - approve ERC20 token
- [x]  Tokens - approve ERC721 token
- [x]  ENS - name resolution
- [x]  Gas fee - EIP-1559 gas
- [x]  Gas fee - legacy gas
- [x]  Sign - ETH sign
- [x]  Sign - personal sign
- [x]  Sign - sign in with Ethereum
- [x]  Sign - sign typed with data (v3 and v4)
- [x]  Swap - swap tokens
- [x]  Incoming transactions history
- [x]  Custom nonce
- [x]  Cancel transaction using custom nonce
- [x]  Stuck multiple transactions (cancel / speed up)
- [x]  PPOM

**Pull request merged**